mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_deserializer.rs

```diff
@@ -0,0 +1,352 @@
+use crate::sm_c8y_mapper::error::SmartRestDeserializerError;
+use csv::ReaderBuilder;
+use json_sm::{SoftwareModule, SoftwareModuleUpdate, SoftwareUpdateRequest};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug)]
+enum CumulocitySoftwareUpdateActions {
+    Install,
+    Delete,
+    UnknownAction,
+}
+
+impl From<String> for CumulocitySoftwareUpdateActions {
```

Comment:
Using `TryFrom` instead of `From` would let us get rid of that `UnknownAction` enum variant and express the failure cases the Rust way with an `Result:Err`